aws/resource_aws_dx_hosted_transit_virtual_interface_test.go

```diff
@@ -0,0 +1,173 @@
+	rName := fmt.Sprintf("tf-testacc-transit-vif-%s", acctest.RandString(9))
+	amzAsn := randIntRange(64512, 65534)
+	bgpAsn := randIntRange(64512, 65534)
+	vlan := randIntRange(2049, 4094)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			testAccAlternateAccountPreCheck(t)
+		},
+		ProviderFactories: testAccProviderFactories(&providers),
+		CheckDestroy:      testAccCheckAwsDxHostedTransitVirtualInterfaceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDxHostedTransitVirtualInterfaceConfig_basic(connectionId, rName, amzAsn, bgpAsn, vlan),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsDxHostedTransitVirtualInterfaceExists(resourceNameHostedVif),
+					testAccCheckAwsDxHostedTransitVirtualInterfaceAccepterExists(resourceNameHostedVifAccepter),
+					resource.TestCheckResourceAttr(resourceNameHostedVif, "name", rName),
+					resource.TestCheckResourceAttr(resourceNameHostedVif, "mtu", "1500"),
+					resource.TestCheckResourceAttr(resourceNameHostedVif, "jumbo_frame_capable", "true"),
+					resource.TestCheckResourceAttr(resourceNameHostedVifAccepter, "tags.%", "0"),
+				),
+			},
+			{
+				Config: testAccDxHostedTransitVirtualInterfaceConfig_updated(connectionId, rName, amzAsn, bgpAsn, vlan),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsDxHostedTransitVirtualInterfaceExists(resourceNameHostedVif),
+					testAccCheckAwsDxHostedTransitVirtualInterfaceAccepterExists(resourceNameHostedVifAccepter),
+					resource.TestCheckResourceAttr(resourceNameHostedVif, "name", rName),
+					resource.TestCheckResourceAttr(resourceNameHostedVifAccepter, "tags.%", "1"),
+					resource.TestCheckResourceAttr(resourceNameHostedVifAccepter, "tags.Environment", "test"),
+				),
+			},
+			// Test import.
+			{
+				Config:            testAccDxHostedTransitVirtualInterfaceConfig_updated(connectionId, rName, amzAsn, bgpAsn, vlan),
+				ResourceName:      resourceNameHostedVif,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckAwsDxHostedTransitVirtualInterfaceDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).dxconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_dx_hosted_transit_virtual_interface" {
+			continue
+		}
+
+		input := &directconnect.DescribeVirtualInterfacesInput{
+			VirtualInterfaceId: aws.String(rs.Primary.ID),
+		}
+
+		resp, err := conn.DescribeVirtualInterfaces(input)
+		if err != nil {
+			return err
+		}
+		for _, v := range resp.VirtualInterfaces {
+			if *v.VirtualInterfaceId == rs.Primary.ID && !(*v.VirtualInterfaceState == directconnect.VirtualInterfaceStateDeleted) {
+				return fmt.Errorf("[DESTROY ERROR] Dx Private VIF (%s) not deleted", rs.Primary.ID)
+			}
+		}
+	}
+	return nil
+}
+
+func testAccCheckAwsDxHostedTransitVirtualInterfaceExists(name string) resource.TestCheckFunc {
```

Comment:
Same note as #8522 here -- the `testAccCheck...Exists()` should be verifying the infrastructure was properly created via the API -- similar but opposite of `testAccCheckAwsDxHostedTransitVirtualInterfaceDestroy` above. 👍

Reply:
Will do.